Allow customizing GoDoc URL and update godoc badge image

alexandregv opened the pull request:
With this PR the GoDoc URL (default https://pkg.go.dev) can now be customized, useful when you host your own instance of [pkgsite](https://pkg.go.dev/golang.org/x/pkgsite).

I also updated the GoDoc badge image with a up-to-date one (godoc.org -> pkg.go.dev).

posener commented on goreadme.go:
```diff
@@ -142,6 +142,8 @@ type Config struct {
 	// github.com/user/project/package or github.com/user/project/version.
 	ImportPath string `json:"import_path"`
+	// GoDocURL is the Go Doc URL used in the GoDoc Badge.
+	GoDocURL string `json:"go_doc_url"`
```
Maybe `godoc_url`, to be consistent with `godoc-url` in `main.go`?

alexandregv replied:
Indeed!

alexandregv commented:
I fixed a bug (missing default value for godoc_url) and added a test (pkg14) to test custom godoc_url.
The CI tests are still failing but it seems its related to my previous PR #127, for which I didn't fixed the tests... I will do it ASAP.
